Refactor host and event schemas

mighabana opened the pull request:
SUMMARY:

This pull request refactors the schema for hosts and events with the following changes:

- splitting the EventHost table into three separate models one for each type of host: Orgs, Offices and Sanggu
- refactor the EventHost model to only contain the name, org_hosts, office_hosts, and sanggu_hosts fields. As of now this model will only have one entity which is the Loyola Schools Hosts. In the future it can be expanded to contain for example Grade School Hosts, High School Hosts, etc.
- delete the model for host_types
- create a modal for org_types (COA, COP)
- refactoring the cluster and org_type fields to be applicable only for the OrgHost type
- add ManyToMany fields for org_hosts, office_hosts and sanggu_hosts in the Event model

Additionally with the changes in the models changes in the fixtures had to be changed accordingly. Likewise changes to the admin models also had to be changed. The changes that were made to the admin side were:

- deleting the host relationship for Events
- cleaning up the fields for EventHosts to take into account deleted fields within the model
- Changing HostInline to OrgInline to be used in the clusters

TEST PLAN:

Test the following endpoints to see if they return the proper responses:

- events/
should still return the same as before except that know multiple hosts are allowed

- event_host/
should return a list of all event_hosts

- org/
should return a list of all orgs

- office/
should return a list of all offices

- sanggu/
should return a list of all sanggu bodies

TO-FIX & LIMITATIONS:
- as of now it is possible to have no hosts when creating an event. This can be tested by creating a fixture where each host field is left empty and the event will still be created. This can be fixed in a later commit I think it's possible by overriding the validations of the serializer.
- admin page models need to be updated to account the changes in the models. This will also be fixed in a later commit.

daniddelrio commented:
A few things I changed:
1. I changed some things in the serializers of Org, Office and Sanggu so that the events in each Detail page will be shown
2. Same goes for the event_host so that the all the hosts of each event_host will be shown in the detail page
3. To make the plural form uniform for the urls, I changed /org and /office to /orgs and /offices.
4. To prevent the error when creating an event in /events, I made event_logistics read_only in the EventSerializer for now
5. The __str__ for OrgType was still type_name so I just changed it to name.

Concerns:
1. I feel that event_host is a bit confusing when referring to if it's under LS, GS or HS. Maybe change it to something like host_level or school_type? I'm not really sure, but I just don't think event_host is a good name for it since this model refers to a broader level of host. 
2. Do we remove the is_accredited field now that LIONS won't be included?
3. If an org wants to post an event with multiple hosts, how will that work? Will they choose the hosts that will host along with them and the chosen hosts have to approve as well as the admin?

Bugs:
1. The image field in the /events post shows up as a text field for me.
<img width="1025" alt="screen shot 2018-08-15 at 10 02 42 pm" src="https://user-images.githubusercontent.com/35568696/44154638-c0a4a834-a0dd-11e8-85aa-fea417d83fbe.png">

Things to do aside from the ones you listed:
1. In the event_view, there's still a query for "host_type_id". Also in list_items it doesn't account for the event_logistics yet.
2. Updating docs based on the refactoring
3. Once the admin part is done include also the events in each host page as well as the hosts for the LS page using inlines also.
I can help with these if you want!

mighabana commented:
Changed the following things given Dani's comments:

- Updated the models to remove the is_accredited field
- Updated the admin page to include inlines for the event_host and all the different types
- Added inlines for each host type for the events
- Created admin models for each of the host types
